test/user.js

```diff
@@ -0,0 +1,106 @@
+const chai = require('chai'),
+  dictum = require('dictum.js'),
+  server = require('./../app'),
+  User = require('./../app/models').user,
+  should = chai.should();
+
+describe('/users POST', () => {
+  it('Successful', done => {
+    chai
+      .request(server)
+      .post('/users')
+      .send({
+        firstName: 'firstName',
+        lastName: 'lastName',
+        username: 'username',
+        password: 'password',
+        email: 'email1@wolox.com.ar'
+      })
+      .then(res => {
+        res.should.have.status(201);
+      })
+      .then(() => done());
+  });
+
+  it('should fail because email is in use', done => {
+    User.createModel({
+      firstName: 'firstName',
+      lastName: 'lastName',
+      username: 'username',
+      password: 'password',
+      email: 'email1@wolox.com.ar'
+    });
+    chai
+      .request(server)
+      .post('/users')
+      .send({
+        firstName: 'firstName',
+        lastName: 'lastName',
+        username: 'username',
+        password: 'password',
+        email: 'email1@wolox.com.ar'
+      })
+      .catch(err => {
+        err.should.have.status(422);
+        err.response.should.be.json;
+        err.response.body.should.have.property('message');
+        err.response.body.should.have.property('internal_code');
+      })
+      .then(() => done());
```

Comment:
remember to put the done inside the catch statement in the failing cases, otherwise if the request ends succesfully it will skip the catch and pass the test.